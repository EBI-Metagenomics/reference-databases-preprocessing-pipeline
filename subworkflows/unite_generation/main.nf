
include { UNCOMPRESS_FILE } from '../../modules/uncompress_file/main.nf'
include { GENERATE_UNITE_TAX } from '../../modules/generate_unite_tax/main.nf'
include { CLEAN_FASTA } from '../../modules/clean_fasta/main.nf'
include { MAKE_OTU_FILE } from '../../modules/make_otu_file/main.nf'
include { GENERATE_MSCLUSTER } from '../../modules/generate_mscluster/main.nf'

workflow UNITE_GENERATION {

    main:

        dummy_fasta = file(params.dummy_fasta)
        unite_fasta = file(params.unite_download_fasta)

        UNCOMPRESS_FILE(
            unite_fasta,
            "unite.fasta"
        )

        GENERATE_UNITE_TAX(
            UNCOMPRESS_FILE.out.uncmp_file,
            params.unite_tax_header,
            params.unite_version,
            params.unite_label
        )

        CLEAN_FASTA(
            UNCOMPRESS_FILE.out.uncmp_file,
            GENERATE_UNITE_TAX.out.tax,
            params.unite_version,
            params.unite_label
        )

        MAKE_OTU_FILE(
            GENERATE_UNITE_TAX.out.tax,
            params.empty_file,
            params.unite_version,
            params.unite_label
        )

        GENERATE_MSCLUSTER(
            dummy_fasta,
            CLEAN_FASTA.out.cleaned_fasta,
            GENERATE_UNITE_TAX.out.tax,
            params.unite_version,
            params.unite_label
        )

    emit:
        fasta = CLEAN_FASTA.out.cleaned_fasta
        tax = GENERATE_UNITE_TAX.out.tax
        otu = MAKE_OTU_FILE.out.otu

}