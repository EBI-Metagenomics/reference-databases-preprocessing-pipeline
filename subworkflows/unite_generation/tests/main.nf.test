nextflow_workflow {

    name "Test Workflow UNITE_GENERATION"
    script "subworkflows/unite_generation/main.nf"
    workflow "UNITE_GENERATION"
    tag "subworkflows"
    tag "unite_generation"

    test("Should run without failures") {

        when {
            params {
                outdir = "$outputDir"
                unite_version = "9.0"
                unite_label = "UNITE"
                unite_download_fasta = "${projectDir}/subworkflows/unite_generation/tests/data/test.fasta.gz"
                unite_tax_header = "${projectDir}/subworkflows/unite_generation/tests/data/unite_tax_header.txt"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out.fasta).match("fasta") },
                { assert snapshot(workflow.out.tax).match("tax") },
                { assert snapshot(workflow.out.otu).match("otu") }
            )
        }

    }

}
