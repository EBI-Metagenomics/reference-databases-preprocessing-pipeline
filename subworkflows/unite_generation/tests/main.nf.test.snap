{
    "otu": {
        "content": [
            [
                "UNITE.otu:md5,c81986c182c1d771d42b9dd3ecb3ed90"
            ]
        ],
        "timestamp": "2024-01-04T11:36:58.70948944"
    },
    "tax": {
        "content": [
            [
                "UNITE-tax.txt:md5,79406418c313fe710d5101faa9000a90"
            ]
        ],
        "timestamp": "2024-01-04T11:36:58.683052616"
    },
    "fasta": {
        "content": [
            [
                "UNITE.fasta:md5,f82eaa0cfa64a100d0c3fb7579e7eca2"
            ]
        ],
        "timestamp": "2024-01-04T11:36:58.621614861"
    }
}